### ISSUE TITLE
project url api? 

### ISSUE THREAD

**Issue by ericwinger** [GemTalk/Rowan#463]
Is there a rowan api to get the remote url of a project? It would be handy to display this on the Project tab. 

This or equivalent - 
```
ewinger@korea:/korea2/users/ewinger/shared/repos/RowanSample1$ git config --get remote.origin.url
https://github.com/dalehenrich/RowanSample1.git
```

**Comment by dalehenrich** [comment]
There is a way to get it ... I'm working in v2.0.0 right now and i've added an official way to get it there ... I can add code to RwProject if you want ...

**Comment by ericwinger** [comment]
I think that it would be useful, especially if one needs to verify they've cloned the right repository. Do you want me to open another issue to request it?

**Comment by dalehenrich** [comment]
Yes ... I've found that the project source is generally useful ... here's the project list from tODE:
![image](https://user-images.githubusercontent.com/623951/50702575-26e68300-1006-11e9-96d0-a8df11812213.png)

I should be able to add this to RwProject shortly .... no need to open another issue ... I'll let you know when it is available ...

**Comment by dalehenrich** [comment]
RwProject>>projectUrl is available on the latest candidateV1.2 branch

**Comment by ericwinger** [comment]
Thank you.

**Comment by dalehenrich** [comment]
@ericwinger ... `RwProject>>projectUrl` is ready when you are ... see latest candidateV1.2 branch

**Comment by ericwinger** [comment]
@dalehenrich - I've updated to the latest candidateV1.2 branch but I'm a little confused by the results I'm seeing. In my ``$ROWAN_PROJECTS_HOME`` directory, I see this: 
```
ewinger@korea:/korea2/users/ewinger/shared/repos/Rowan$ git config --get remote.origin.url
git@github.com:GemTalk/Rowan.git
```
but from my Jadeite image, I see: 
```
(RwProject newNamed: 'Rowan') projectUrl 'https://github.com/dalehenrich/Rowan'
```
Rowan projects home is set correctly: 
```
System gemEnvironmentVariable: 'ROWAN_PROJECTS_HOME' '/export/korea2/users/ewinger/shared/repos/'
```

Why wouldn't I see ``projectUrl`` be from the GemTalk account?

**Comment by dalehenrich** [comment]
I think we have a failure to communicate ... I know you don't look at project specs often, but what I gave you is what I have been calling a  `projectUrl` - note the `#projectUrl` field in the load specification:
```ston
 RwSimpleProjectSpecification{
	#specName : 'Rowan',
	#version : '0.2.0',
	#projectUrl : 'https://github.com/dalehenrich/Rowan',
	#repoSpec : RwGitRepositorySpecification {
		#committish : 'master',
		#committishType : 'branch'
	},
	#configsPath : 'rowan/configs',
	#repoPath : 'rowan/src/',
	#specsPath : 'rowan/specs',
	#defaultConfigurationNames : [ 'Load' ],
	#defaultGroupNames : [ 'core', 'tests', 'jadeServer' ],
	#comment : 'JadeServer Rowan load specification'
}
```
What you are asking for is what I have been calling a `remoteUrl`in the v2.0.0 work. 

I've found the `projectUrl` to be sufficient to provide a pointer the github/gitlab/etc/ site where the project is housed ... 

In v2.0.0 the `remoteUrl` will be (optionally) defined in the load spec and will be indicative of the remote url used to create the repository.

When it comes to remote urls, are you aware that a repository can have multiple remote urls associated with it? Here's one of my projects with multiple *remotes*:
```
rogue:smalltalkCI>git remote -v
hpi	https://github.com/hpi-swa/smalltalkCI.git (fetch)
hpi	https://github.com/hpi-swa/smalltalkCI.git (push)
origin	git@github.com:dalehenrich/smalltalkCI.git (fetch)
origin	git@github.com:dalehenrich/smalltalkCI.git (push)
rogue:smalltalkCI>git config --get remote.origin.url
git@github.com:dalehenrich/smalltalkCI.git
rogue:smalltalkCI>git config --get remote.hpi.url
https://github.com/hpi-swa/smalltalkCI.git
```
It is a good idea to allow for management of remote urls, but I think that to display only the `origin` is not the right way to go about it ... until we allow the user to set the `name` of the remote that they want to use for `push` and `pull` operations and when we allow that, then it would be sufficient to simply display the `name` rather than the remote url itself (assuming that the user will be able to look at the full list of remote urls associated with the project) ...

For the short term it is probably good enough to display the `project url` and wait until we get to the oint where we build a GUI for interacting with git to get into remote urls and the like ...

Give me a call if you want to discuss this some more ...

**Comment by dalehenrich** [comment]
The reason that you are not seeing `GemTalk`is because I haven't updated the project url in the load spec since we've moved the project from `dalehenrich/Rowan` to `GemTalk/Rowan` ... and GithHub automatically forwards `https://github.com/dalehenrich/Rowan` to `https://github.com/GemTalk/Rowan` .. and **that** should be changed:)

**Comment by dalehenrich** [comment]
... branch `candidateV1.2` has been updated to use the correct project url for the Rowan load spec ...

**Comment by dalehenrich** [comment]
The spec is available in v1.2.x, but let's hold off on a GUI for the load spec until you start working on v2.0.0 support ... I'm actually moving away from the spec (still working on exact details) ... the spec **will** still exist on disk, but the in-image data structure is likely to not be a specification as we know it today in v1.2:) and of course there will be other things that will **need** to be added to the project browser (nested configurations, etc.)

**Comment by ericwinger** [comment]
Understood, I'll wait until 2.0.0
